lib/gpfpiso.gi

```diff
@@ -888,5 +964,992 @@ function( G, series )
+  fi;
+  return j;
+end);
+
+BindGlobal("WeylGroupFp",function(ser,n)
+local f,rels,i,j,gens;
+  f:=FreeGroup(List([1..n],x->Concatenation("s",String(x))));
+  gens:=GeneratorsOfGroup(f);
+  rels:=List(gens,x->x^2);
+  if ser="A" then
+    for i in [1..n] do
+      for j in [1..i-1] do
+        if i-j>1 then
+          Add(rels,(gens[i]*gens[j])^2);
+        else
+          Add(rels,(gens[i]*gens[j])^3);
+        fi;
+      od;
+    od;
+  elif ser="B" or ser="C" then
+    for i in [1..n] do
+      for j in [1..i-1] do
+        if i-j>1 then
+          Add(rels,(gens[i]*gens[j])^2);
+        elif j=1 then
+          Add(rels,(gens[i]*gens[j])^4);
+        else
+          Add(rels,(gens[i]*gens[j])^3);
+        fi;
+      od;
+    od;
+  elif ser="D" and n>=4 then
+    Add(rels,(gens[1]*gens[2])^2);
+    Add(rels,(gens[1]*gens[3])^3);
+    Add(rels,(gens[2]*gens[3])^3);
+    for i in [4..n] do
+      for j in [1..i-1] do
+        if i-j>1 then
+          Add(rels,(gens[i]*gens[j])^2);
+        else
+          Add(rels,(gens[i]*gens[j])^3);
+        fi;
+      od;
+    od;
+  elif ser="E" then
+    Add(rels,(gens[1]*gens[2])^3);
+    Add(rels,(gens[2]*gens[3])^3);
+    Add(rels,(gens[3]*gens[4])^3);
+    Add(rels,(gens[3]*gens[5])^3);
+    Add(rels,(gens[5]*gens[6])^3);
+
+    Add(rels,(gens[1]*gens[3])^2);
+    Add(rels,(gens[1]*gens[4])^2);
+    Add(rels,(gens[2]*gens[4])^2);
+    Add(rels,(gens[1]*gens[5])^2);
+    Add(rels,(gens[2]*gens[5])^2);
+    Add(rels,(gens[4]*gens[5])^2);
+    Add(rels,(gens[1]*gens[6])^2);
+    Add(rels,(gens[2]*gens[6])^2);
+    Add(rels,(gens[3]*gens[6])^2);
+    Add(rels,(gens[4]*gens[6])^2);
+    if n>=7 then
+      Add(rels,(gens[6]*gens[7])^3);
+      for i in [1..5] do
+        Add(rels,(gens[i]*gens[7])^2);
+      od;
+    fi;
+    if n=8 then
+      Add(rels,(gens[7]*gens[8])^3);
+      for i in [1..6] do
+        Add(rels,(gens[i]*gens[7])^2);
+      od;
+    elif n>8 then 
+      Error("E>8 does not exist");
+    fi;
+  elif ser="F" and n=4 then
+    Add(rels,(gens[1]*gens[2])^3);
+    Add(rels,(gens[2]*gens[3])^4);
+    Add(rels,(gens[3]*gens[4])^3);
+
+    Add(rels,(gens[1]*gens[3])^2);
+    Add(rels,(gens[1]*gens[4])^2);
+    Add(rels,(gens[2]*gens[4])^2);
+  elif ser="G" and n=2 then
+    Add(rels,(gens[1]*gens[2])^6);
+  else
+    Error("series ",ser," not yet done");
+  fi;
+  return f/rels;
+end);
+
+BindGlobal("IsomorphismFpGroupForWeyl",
+function(G)
+local iso,n,fn,sz,bigcount,tryweyl;
+
+  tryweyl:=function(ser,n)
+  local H,isp,P,iso;
+    H:=WeylGroupFp(ser,n);
+    isp:=IsomorphismPermGroup(H);
+    P:=Image(isp,H);
+    iso:=IsomorphismGroups(G,P);
+    if iso<>fail then
+      P:=List(GeneratorsOfGroup(H),
+        x->PreImagesRepresentative(iso,ImagesRepresentative(isp,x)));
+      iso:=GroupHomomorphismByImagesNC(G,H,P,GeneratorsOfGroup(H));
+    fi;
+    return iso;
+  end;
+
+  # try to identify Weyl series
+  n:=0;
+  repeat
+    n:=n+1;
+    fn:=Factorial(n);
```

Comment:
It probably doesn't matter for any group size we see in practice, but you could also init `fn:=1` before the loop and then write here:
```suggestion
    fn:=fn*n;
```

Reply:
Done